tests/types/index.ts

```diff
@@ -0,0 +1,98 @@
+/*
+* Copyright 2021 Mia srl
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import {
+  getMetrics,
+  FMClientBuilder,
+  KafkaConfig,
+  ConsumerConfig,
+  ProducerConfig,
+  CommandExecutor,
+  CommandErrorHandler
+} from '../../index'
+
+const metrics = {foo: 'bar'}
+getMetrics(metrics)
+
+const log = {foo: 'bar'}
+
+const kafkaConfig: KafkaConfig = {
+  clientId: 'client-id',
+  brokers: 'brokers',
+  authMechanism: 'auth-mechanism',
+  username: 'username',
+  password: 'password',
+  connectionTimeout: 1,
+  authenticationTimeout: 1,
+}
+
+const ClientBuilder = new FMClientBuilder(log, kafkaConfig)
+console.log(ClientBuilder.log)
```

Comment:
Instead of test with console.log, we should use something like https://github.com/SamVerschueren/tsd to test the expected types